src/ucp/core/ucp_request.inl

```diff
@@ -780,17 +802,59 @@ ucp_request_get_memory_type(ucp_context_h context, const void *address,
     return param->memory_type;
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_send_request_init_local_id(ucp_request_t *req)
+{
+    ucp_worker_h worker = req->send.ep->worker;
+    int indirect        = ucp_ep_use_indirect_id(req->send.ep);
+    ucs_status_t status;
+
+    ucp_request_check_local_id_invalid(req);
+    status = ucs_ptr_map_put(&worker->ptr_map, req, indirect,
+                             &req->local_req_id);
+    ucs_assertv_always(status == UCS_OK, "%p: failed to get local id", req);
+}
+
 static UCS_F_ALWAYS_INLINE ucs_ptr_map_key_t
-ucp_send_request_get_id(ucp_request_t *req)
+ucp_send_request_get_local_id(const ucp_request_t *req)
 {
-    return ucp_worker_get_request_id(req->send.ep->worker, req,
-                                     ucp_ep_use_indirect_id(req->send.ep));
+    ucp_request_check_local_id_valid(req);
+    return req->local_req_id;
 }
 
 static UCS_F_ALWAYS_INLINE void
-ucp_send_request_set_id(ucp_request_t *req)
+ucp_send_request_del_local_id(ucp_request_t *req)
```

Comment:
pls rename this function to ucp_request_id_release()

Reply:
done